consumer/consumer.go

```diff
@@ -169,7 +170,9 @@ func (c consumerGroupHandler) ConsumeClaim(sess sarama.ConsumerGroupSession, cla
 			}
 			break
 		}
-		sess.MarkMessage(msg, "")
+		if mark {
```

Comment:
Here we are not considering `mark` if there is a context error, is this the intended behaviour? (to not mark a message even if it's defined as to be marked)

Reply:
I kept it as it is to be backwards compatible and I think it's good behaviour not to mark the message as read when server is shutting down. Other option that could be less misleading is not to call `Discarded` when there is a context error. Do you think that way is more clear.